dashboard/config/routes.rb

```diff
@@ -268,6 +268,8 @@
       end
     end
 
+    get 'projects/for_level/:level_id', to: 'projects#get_or_create_for_level'
```

Comment:
A comment in the function implementation is probably more important, but it seems like more comments in `routes.rb` might also help?